src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/MaxWeather/MaxWeatherCommandHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿using ExadelMentorship.BusinessLogic.Interfaces;
+using ExadelMentorship.BusinessLogic.Models;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ExadelMentorship.BusinessLogic.Features.WeatherFeature.MaxWeather
+{
+    public class MaxWeatherCommandHandler : ICommandHandler<MaxWeatherCommand>
+    {
+        IRWOperation _rwOperation;
+        private readonly IWeatherApiService _weatherApiService;
+        private readonly IConfiguration _configuration;
+
+        public MaxWeatherCommandHandler(IRWOperation rwOperation, IWeatherApiService weatherApiService, IConfiguration configuration)
+        {
+            _rwOperation = rwOperation;
+            _weatherApiService = weatherApiService;
+            _configuration = configuration;
+        }
+
+        public async Task Handle(MaxWeatherCommand command)
+        {
+            _rwOperation.WriteLine("Please enter the cities:");
+            var cities = this.GetCitiesFromInput();
+
+            var tasks = new List<Task<MaxTempCityInfo>>();
+            foreach (var city in cities)
+            {
+                tasks.Add(_weatherApiService.GetTemperatureByCityNameForMaxTemp(city));
+            }
+
+            await Task.WhenAll(tasks);
+            var maxTempCityInfoList = new List<MaxTempCityInfo>();
+            foreach (var task in tasks)
+            {
+                maxTempCityInfoList.Add(task.Result);
+            }
+            
+            var maxTempCityInfo = maxTempCityInfoList.OrderByDescending(x => x.Temperature).First();
+
+            _rwOperation.WriteLine($"City with the highest temperature {maxTempCityInfo.Temperature} C: {maxTempCityInfo.Name}");
+            
+            var statistic = bool.Parse(_configuration["Statistic"]);
+            if (statistic)
+            {
+                if (maxTempCityInfo.ErrorMessage is null)
+                {
+                    _rwOperation.WriteLine($"City: {maxTempCityInfo.Name}. {maxTempCityInfo.Temperature}. Timer:{maxTempCityInfo.DurationTime}");
+                }
+                else
+                {
+                    _rwOperation.WriteLine($"City: {maxTempCityInfo.Name}. Error: {maxTempCityInfo.ErrorMessage}");
+                }
+            }
+                
+        }
+
+        private List<string> GetCitiesFromInput()
+        {
+            var cities = new List<string>();
+            var inputedLine = _rwOperation.ReadLine();
+            string[] words = inputedLine.Split(',');
+            foreach (string word in words)
+            {
+                cities.Add(word.Trim());
+            }
+            return cities;
```

Comment:
Then, this method simplifies to
```
return _rwOperation.ReadLine().Split(',')
```
which makes this method unnecessary. Consider inlining it

Reply:
but, between commas may also be spaces, and I need to parse(trim) each of  them after splitting, that's why I can't make it inline.